packages/style-engine/src/styles/background/index.ts

```diff
@@ -8,19 +8,21 @@ const backgroundImage = {
 	name: 'backgroundImage',
 	generate: ( style: Style, options: StyleOptions ) => {
 		const _backgroundImage = style?.background?.backgroundImage;
-		if (
-			typeof _backgroundImage === 'object' &&
-			_backgroundImage?.source === 'file' &&
-			_backgroundImage?.url
-		) {
+		if ( typeof _backgroundImage === 'object' && _backgroundImage?.url ) {
+			const url = !! options?.baseUrl
```

Comment:
So this was mainly so we wouldn't have to add any ugly code to `useBlockProps`. 

In a previous commit I was trying out modifying the style attribute object:

https://github.com/WordPress/gutenberg/pull/60578/commits/4a0d1075ececa5237c164ed6fe36a2ec31af3f80#diff-292053c9ec98ebdd73943767587e8093315ad3a2bf01ca3b5352efc48326becdR322